src/libs/OptionsListUtils.js

```diff
@@ -365,6 +367,36 @@ function getAllReportErrors(report, reportActions) {
     return allReportErrors;
 }
 
+/**
+ * Get the last message text from the report directly or from other sources for special cases.
+ * @param {Object} report
+ * @returns {String}
+ */
+function getLastMessageTextForReport(report) {
+    let lastMessageTextFromReport = '';
+    if (ReportUtils.isReportMessageAttachment({text: report.lastMessageText, html: report.lastMessageHtml})) {
+        lastMessageTextFromReport = `[${Localize.translateLocal('common.attachment')}]`;
+    } else {
+        lastMessageTextFromReport = report ? report.lastMessageText || '' : '';
+
+        // Yeah this is a bit ugly. If the latest report action that is not a whisper has been moderated as pending remove, then set the last message text to the text of the latest visible action that is not a whisper.
+        const lastNonWhisper =
+            _.find(allSortedReportActions[report.reportID], (action) => {
+                const isWhisper = (action.whisperedTo || []).length > 0;
+                return !isWhisper;
+            }) || {};
+        if (lodashGet(lastNonWhisper, 'message[0].moderationDecisions[0].decision') === CONST.MODERATION.MODERATOR_DECISION_PENDING_REMOVE) {
+            const latestVisibleAction =
+                _.find(allSortedReportActions[report.reportID], (action) => {
+                    const isWhisper = (action.whisperedTo || []).length > 0;
+                    return ReportActionUtils.shouldReportActionBeVisible(action, action.reportActionID) && !isWhisper;
+                }) || {};
+            lastMessageTextFromReport = lodashGet(latestVisibleAction, 'message[0].text', '');
+        }
```

Comment:
Right now, we're doing this for every report. 
Since this flow is not the main flow can we update the logic so that by default, we don't look for the last non whisper unless we have to?

So basically

Check the last action had a moderation decision

If not, proceed as normal
Otherwise, find the last visible reportAction

Reply:
We need to check the last action that is not a whisper, because otherwise we're looking to see if the whisper is moderated which will always be false.

_.find will stop as soon as it finds a non whisper action, so it's only going to look through n actions where n is the number of whispers in a row from the latest action moving back in time. I think that's ok.

Reply:
Hmm yeah, I guess there isn't a great way to avoid this